### DISCUSSION TITLE
Next js CORS not working

### DISCUSSION THREAD

**Discussion by adityar15** [vercel/next.js#33168] [Help]
This is what my API code looks like 

```

import Cors from 'cors'

function initMiddleware(middleware) {
    return (req, res) =>
      new Promise((resolve, reject) => {
          
       middleware(req, res, (result) => {
         
          if (result instanceof Error) {
            return reject(result)
          }
          return resolve(result)
        }) 
      })
  }


  const cors = initMiddleware(
    Cors({
        "methods": ["POST"]
      })
  )

export default async function handler(req, res) {

    await cors(req, res)
    res.status(200).json({ name: req.query.name, hobby: req.query.name })
 
  
  }
  


```

now when I make a get request from the same next app's full-page component, it still goes through well rather than throwing an error. Is it the correct behaviour?

**Comment by Dupflo** [comment]
Same thing for me, POST seems to does not work

**Reply by omkarajagunde** [reply]
Yes POST is not working for me as well

**Reply by icyJoseph** [reply]
This might be a silly question, but if the page is in the same domain as the `API` route, why would there be a CORS trigger? CORS is for when domain A, makes a request to domain B. You can see that there's no preflight request on your network tab in the DevTools, when the request comes from domain A to domain A.

It also looks to me like, one has to explicitly pass the allowed origins, otherwise, it defaults to '*', aka, allows every domain B type, to make requests.

Can anyone clarify if I have misread the issue?
